### ISSUE TITLE
Web-based client

### ISSUE THREAD

**Issue by antonmedv** [deployphp/deployer#350]
Web-based client for deploy by button.

**Comment by nepsdotin** [comment]
Would like to build up the web feature. How do i Start ? How to go about ?

**Comment by oanhnn** [comment]
@nepsdotin Please fork this repo, create a branch base branch 4.x and make pull request after you finished

**Comment by elfet** [comment]
This will be really cool!

**Comment by elfet** [comment]
Please use ES6, babel, webpack, react or monkberry.

**Comment by Kleinast** [comment]
Hi,

I am very interest in this feature, and I may want contribute to this projet, depend on what you plan to do as feature for this web interface. Can you details a little what you want to do ?

Thx

**Comment by elfet** [comment]
I want create simple command for weи server:
```
dep server
```

And also web phar for working under nginx and other servers.

**Comment by elfet** [comment]
I'm going to implement web app by myself.

**Comment by barryvdh** [comment]
I like this idea :)

Do you have a rough idea yet? I would say it doesn't have to be complicated, just a web UI where you can add/edit projects. And just define which recipe the project uses, ability to create/edit a serverlist yaml file. Then just run the recipe with the correct serverList file for a given project for the chosen environments.
Would just need to be able to run the tasks from a PHP project instead of CLI, but that shouldn't be to hard. And parse the serverList to know which environments are possible, and perhaps even the recipe to know the tasks..

Showing the output in 'realtime' would be a bit trickier perhaps, but for the rest wouldn't need so much Javascript, right?
(Or am I thinking to easy about this? ;))

**Comment by elfet** [comment]
I'm thinking about gui to running task. No edit functionality for first release.

**Comment by barryvdh** [comment]
I tried some things, and it is relatively doable with Laravel, using queues and the Symfony Process command. Just web doesn't really work nicely with forward agent.

Not really sure if you're interested, but there already seems to be a working interface here: https://github.com/ngmy/webloyer by @ngmy

**Comment by pluseg** [comment]
@elfet is it still actual feature request?

**Comment by elfet** [comment]
Yes, and i'm still working on it. This is really big feature, but i'm really want to ship it as fast as possible. Now don't have enough time, due to main work.

**Comment by tuupola** [comment]
FWIW I think this should be a separate repo. It will bloat `dep` for those who are capable of using cli.

**Comment by antonmedv** [comment]
@tuupola of course

**Comment by sprankhub** [comment]
Thanks for that cool tool @antonmedv! Is there any update regarding the web interface?

**Comment by antonmedv** [comment]
Working hard on it)

**Comment by ErikDohmen** [comment]
Hi @antonmedv,
Any news on the progress of the gui. I would love to test it for several sites and where possible/ needed contribute.

Any chance to get access to the current version and test that?

**Comment by antonmedv** [comment]
Hi @ErikDohmen 

I'm working on it :) But now can't assign you to beta testers program as we are migrating to new servers.

**Comment by lucasdcrk** [comment]
Still no news? :)

**Comment by antonmedv** [comment]
Unfortunately no news, progress is very slow as I have only free time and main work.  Currently rewriting backend for easy worker installation of deployment.

**Comment by ErikDohmen** [comment]
Is there something I can do to help? I'm willing to make some time to 
dive into to help if needed.

Sobit (?} {!)
Tailormade software
Agro Business Park 75-83
6708 PV  Wageningen
026 800 10 22
06 45 194 596
Erik@sobit.nl
http://www.sobit.nl

**Comment by antonmedv** [comment]
@ErikDohmen currently no, because I can't publish code yet.

**Comment by sprankhub** [comment]
Why has this been closed?

**Comment by antonmedv** [comment]
No need for an issue. :)

**Comment by sprankhub** [comment]
Why not? :)

**Comment by antonmedv** [comment]
I'm in the process of cleaning issues on GitHub, starting from old ones. I decided this one not needed to be opened, as this is more like a todo list for me.

**Comment by VivekShingala** [comment]
Hi @antonmedv 

Is this still in your todo list?

**Comment by antonmedv** [comment]
Maybe as a paid extension. Would you pay for such feature?

**Comment by Schrank** [comment]
From my point of view, this depends on the price. But most people using deployer most likely use github actions or bitbucket/gitlab pipelines ... which somehow includes a deploy button by running the pipeline.

TL;DR we don't need it.
